ground/src/test/java/com/google/android/ground/repository/SurveyRepositoryTest.kt

```diff
@@ -17,106 +17,88 @@ package com.google.android.ground.repository
 
 import com.google.android.ground.BaseHiltTest
 import com.google.android.ground.coroutines.DefaultDispatcher
-import com.google.android.ground.model.Survey
-import com.google.android.ground.persistence.local.LocalDataStore
-import com.google.android.ground.persistence.local.LocalDataStoreModule
 import com.google.android.ground.persistence.local.LocalValueStore
-import com.google.android.ground.persistence.local.room.RoomLocalDataStore
+import com.google.android.ground.persistence.local.room.LocalDatabase
 import com.google.android.ground.persistence.local.stores.LocalSurveyStore
 import com.google.common.truth.Truth.assertThat
 import com.sharedtest.FakeData.SURVEY
 import com.sharedtest.persistence.remote.FakeRemoteDataStore
-import dagger.hilt.android.testing.BindValue
 import dagger.hilt.android.testing.HiltAndroidTest
-import dagger.hilt.android.testing.UninstallModules
-import io.reactivex.Completable
-import io.reactivex.Maybe
 import java8.util.Optional
 import javax.inject.Inject
+import kotlin.test.assertFails
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.rx2.await
 import kotlinx.coroutines.test.advanceUntilIdle
 import kotlinx.coroutines.test.runTest
 import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
-import org.mockito.ArgumentMatchers.anyString
-import org.mockito.InjectMocks
-import org.mockito.Mock
 import org.mockito.Mockito.*
-import org.mockito.kotlin.any
 import org.robolectric.RobolectricTestRunner
 
 @OptIn(ExperimentalCoroutinesApi::class)
-@UninstallModules(LocalDataStoreModule::class)
 @HiltAndroidTest
 @RunWith(RobolectricTestRunner::class)
 class SurveyRepositoryTest : BaseHiltTest() {
-  @BindValue @InjectMocks var mockLocalDataStore: LocalDataStore = RoomLocalDataStore()
-
-  @BindValue @Mock lateinit var mockLocalSurveyStore: LocalSurveyStore
-
+  @Inject lateinit var localDatabase: LocalDatabase
+  @Inject lateinit var surveyStore: LocalSurveyStore
   @Inject lateinit var fakeRemoteDataStore: FakeRemoteDataStore
-
   @Inject lateinit var surveyRepository: SurveyRepository
-
   @Inject lateinit var localValueStore: LocalValueStore
-
   @DefaultDispatcher @Inject lateinit var testDispatcher: CoroutineDispatcher
 
   @Before
   override fun setUp() {
     super.setUp()
-    `when`(mockLocalSurveyStore.insertOrUpdateSurvey(any())).thenReturn(Completable.complete())
+    // Reset local test db between tests.
+    localDatabase.clearAllTables()
```

Comment:
Isn't this implicit?

Reply:
I suppose it is. Each test gets a different instance of `LocalDatabase`, which I assume has a different SQLite behind it. Thanks for pointing this out.